### ISSUE TITLE
Add a benchmark printing header

### ISSUE THREAD

**Issue by e-harvey** [ofi-cray/cray-tests#1]
We'd like two output types for the benchmarks 1) tabular and 2) csv.
- The printing header should offer two mechanisms of specifying the output format:
  1) Pull the format from an exisiting csv file
  2) Let the user specify the format manually

When calling the print routine the output will always appear in tabular format in stderr and only be appended to a given csv file if the file exists.

**Comment by e-harvey** [comment]
The benchmarks will only be responsible for collecting data and interfacing with the print utility.  For example, the print utility could be used as follows:

``` c
double *arr = init_info("foobar", "foobar.csv", "bandwidth(MB), runtime(s)", 2);
char *params = "iterations = 1024, max_msg_size = 2048";

arr[0] = calc_bw(1024, 2048);
arr[1] = rt();
print_data(params);

fini_info();
```

# Proposed interfaces

``` c
/**                                                                                                                                                                                                                                          
 * This function must be called before print_data() and fini_info().                                                                                                                                                                         
 *                                                                                                                                                                                                                                           
 * If the csv_path is not valid or the csv_path points to an empty file then                                                                                                                                                                 
 * the header must be non-null.  The csv file will only be written to by                                                                                                                                                                     
 * print_data() if it exists.                                                                                                                                                                                                                
 *                                                                                                                                                                                                                                           
 * @param test_name  the name of the test.                                                                                                                                                                                                   
 * @param csv_path   the full or relative path to the csv file.                                                                                                                                                                              
 * @param csv_header the comma separated list of column headings.                                                                                                                                                                            
 * @param data_len   the length of the pointer returned.                                                                                                                                                                                     
 *                                                                                                                                                                                                                                           
 * @return a pointer to the data buffer the user will append to.                                                                                                                                                                             
 */
static inline double *init_info(char *test_name, char *csv_path,
                                char *csv_header, size_t data_len);
```

``` c
/**                                                                                                                                                                                                                                          
 * Append the data to stderr and the csv_file if it exists.                                                                                                                                                                                  
 *                                                                                                                                                                                                                                           
 * @param params the parameters used to collect the line of data.                                                                                                                                                                            
 */
static inline void print_data(char *params);
```

``` c
/**                                                                                                                                                                                                                                          
 * Should be called before exiting the benchmark. Cleans up                                                                                                                                                                                  
 * globals and memory allocated by init_info.                                                                                                                                                                                                
 */
static inline void fini_info()
```

@sungeunchoi @jswaro

**Comment by jswaro** [comment]
I'm not certain how this works or prints data based on the prototype. Could you elaborate?

**Comment by e-harvey** [comment]
> I'm not certain how this works or prints data based on the prototype. Could you elaborate?

Sure, the steps would be:

1) The user calls `init_info() before running their benchmarks.`init_info()` allocates an array of doubles, initializes the test information, tries to open the csv file, and returns a pointer to the array of doubles to the caller.

2) The user runs their benchmark and fills in the array of doubles, when the the array is full the user should call `print_data()` which will print a line of output to stderr and the csv file (if it exists).

3) Repeat 2) as many times as needed.

4) `fini_info()` is called after the benchmark is done running, it will reset states, close the csv file if it was opened, and free memory.

I forgot to mention earlier that I plan on having `print_data()` write the date and time to the first collumn of the csv file every time it's called.

**Comment by e-harvey** [comment]
This has been addressed by #6.
